testkube: init at 2.1.56

mathstlouis opened the pull request:
Added Testkube CLI at version 2.1.56

> Testkube is a kubernetes-native framework for test definition and execution.
> The Testkube CLI allows you to both manage your Testkube installation and interact with your Testkube resources

refs.:
https://docs.testkube.io/articles/install/cli
https://github.com/kubeshop/testkube

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mathstlouis committed:
testkube: init at 2.1.56

Aleksanaa commented on pkgs/by-name/te/testkube/package.nix:
```diff
@@ -0,0 +1,36 @@
+  doCheck = false;
+  subPackages = [ "cmd/kubectl-testkube" ];
+
+  meta = with lib; {
+    description = "Kubernetes-native framework for test definition and execution";
+    homepage = "https://github.com/kubeshop/testkube/";
+    license = licenses.mit;
+    mainProgram = "kubectl-testkube";
+    maintainers = with maintainers; [ ];
```
```suggestion
  meta = {
    description = "Kubernetes-native framework for test definition and execution";
    homepage = "https://github.com/kubeshop/testkube/";
    license = lib.licenses.mit;
    mainProgram = "kubectl-testkube";
    maintainers = with lib.maintainers; [ ];
```

mathstlouis replied:
ok fixed in 8b389a095b1cc04db34d2f6f0e5e524270ba7e04

Aleksanaa commented on pkgs/by-name/te/testkube/package.nix:
```diff
@@ -0,0 +1,36 @@
+  doCheck = false;
+  subPackages = [ "cmd/kubectl-testkube" ];
+
+  meta = with lib; {
+    description = "Kubernetes-native framework for test definition and execution";
+    homepage = "https://github.com/kubeshop/testkube/";
+    license = licenses.mit;
+    mainProgram = "kubectl-testkube";
+    maintainers = with maintainers; [ ];
```
new packages must have maintainers.

mathstlouis replied:
fixed in 8b389a095b1cc04db34d2f6f0e5e524270ba7e04

Aleksanaa commented on pkgs/by-name/te/testkube/package.nix:
```diff
@@ -0,0 +1,36 @@
+    owner = "kubeshop";
+    repo = "testkube";
+    rev = "v${version}";
+    sha256 = "sha256-P+A9lUMzQ3M0SEVZBMDSMj8S0uCsEhadv5vDRxbQORA=";
```
```suggestion
    hash = "sha256-P+A9lUMzQ3M0SEVZBMDSMj8S0uCsEhadv5vDRxbQORA=";
```

mathstlouis replied:
fixed in e25f98f28ddc2fc7ced25ce5bcde8e84293168a2

Aleksanaa commented on pkgs/by-name/te/testkube/package.nix:
```diff
@@ -0,0 +1,36 @@
+    "-X main.date=1970-01-01-00:00:01"
+  ];
+
+  doCheck = false;
```
elaborate why you disabled the test in comment, for example (just imaginary)

```suggestion
  doCheck = false; # Require binding to kubernets service port
```

mathstlouis replied:
Fixed in feb8a2753c1f2c9fd5a35edc9b4fd26c293ac1e4. Don't remember why I skipped checks in my personal configs. Build works with check phase now.

Aleksanaa commented on pkgs/by-name/te/testkube/package.nix:
```diff
@@ -0,0 +1,36 @@
+
+  doCheck = false;
+  subPackages = [ "cmd/kubectl-testkube" ];
+
```
Would be better to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook, since you override version and skipped test.

mathstlouis replied:
I just cannot get the versioncheckhook working with testkube. When running 'kubectl-testkube version' it displays some colored ASCII art before displaying the version so I think this might mess up the version parser.

Aleksanaa replied:
Could you try https://nixos.org/manual/nixpkgs/unstable/#tester-testVersion?

mathstlouis replied:
@Aleksanaa sorry I finally figured it out: 'kubectl-testkube version' command needs to reach out to 'https://api.github.com/repos/kubeshop/testkube/releases/latest' else it fails without displaying version. Thus I cannot use the versioncheckhook and testers.testVersion:

```
❯ nix-build --attr pkgs.testkube.passthru.tests                                       
this derivation will be built:
  /nix/store/yhly75lw5107lksdlh3yld4ykzyzn8p6-testkube-2.1.56-test-version.drv
building '/nix/store/yhly75lw5107lksdlh3yld4ykzyzn8p6-testkube-2.1.56-test-version.drv'...
kubectl-testkube version returned a non-zero exit code.

Gettong latest version (error: Get "https://api.github.com/repos/kubeshop/testkube/releases/latest": dial tcp: lookup api.github.com: no such host)
error: builder for '/nix/store/yhly75lw5107lksdlh3yld4ykzyzn8p6-testkube-2.1.56-test-version.drv' failed with exit code 1;
       last 3 log lines:
       > kubectl-testkube version returned a non-zero exit code.
       >
       > Gettong latest version (error: Get "https://api.github.com/repos/kubeshop/testkube/releases/latest": dial tcp: lookup api.github.com: no such host)
       For full logs, run 'nix log /nix/store/yhly75lw5107lksdlh3yld4ykzyzn8p6-testkube-2.1.56-test-version.drv'.
```

Do you know of a workaround for this? Thank you!

Aleksanaa replied:
I don't think there are either

mathstlouis committed:
testkube: removed with lib + added mathstlouis as maintainer

mathstlouis committed:
testkube: fetchFromGitHub: use hash instead of sha256

mathstlouis committed:
testkube: do not skip check phase

mathstlouis commented:
@Aleksanaa Thanks for your time reviewing this PR.

I think I addressed all your suggestions. I just cannot get the versioncheckhook working with testkube. When running 'kubectl-testkube version' it displays some colored ASCII art before displaying the version so I think this might mess up the version parser.
